java/libs/src/main/java/org/physical_web/collection/PhysicalWebCollection.java

```diff
@@ -211,4 +216,65 @@ public void jsonDeserialize(JSONObject jsonObject) throws PhysicalWebCollectionE
       addMetadata(pwsResult);
     }
   }
+
+  /**
+   * Return a list of PwPairs sorted by rank in descending order.
+   * @param dedupSiteUrls whether or not to dedup PwPairs by result urls.
+   * @param minRank the minimum rank to keep in the list.
+   * @return a sorted list of PwPairs.
+   */
+  public List<PwPair> getPwPairsSortedByRank(boolean dedupSiteUrls, double minRank) {
```

Comment:
I like this proposal.  Does PwResult.devices contain all UrlDevices for a particular broadcastUrl or siteUrl?  I would lean toward siteUrl since grouping by broadcastUrl isn't as useful.